src/main/java/mate/academy/rickandmorty/service/client/RickAndMortyClient.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.rickandmorty.service.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.rickandmorty.dto.external.CharacterResponseDto;
+import mate.academy.rickandmorty.dto.external.CharacterResultsResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RickAndMortyClient {
+    private static final String BASE_URL = "https://rickandmortyapi.com/api/character?page=%s";
+    private static final int PAGES = 42;
+    private final ObjectMapper objectMapper;
+
+    public RickAndMortyClient(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public List<CharacterResultsResponseDto> getAllCharacters() {
+        HttpClient httpClient = HttpClient.newHttpClient();

```

Comment:
1 Consider making HttpClient the spring bean and inject it directly as a class level field. 

2 Make it class level field -> it shouldn't be instantiated every method invokation

```suggestion
        HttpClient httpClient = HttpClient.newHttpClient();

```